offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -802,6 +802,230 @@ export class CkbTxGenerator extends CkbTxHelper {
+
+    let totalSudtOutputCapacity = BigInt(0);
+    let totalSudtOutputAmount = BigInt(0);
+    // add recipient outputs
+    const sudtCapacity = BigInt(ForceBridgeCore.config.ckb.sudtSize * 10 ** 8);
+    records.forEach((record) => {
+      const amount = BigInt(record.amount);
+      const recipientCell: Cell = {
+        cell_output: {
+          capacity: '0x0',
+          lock: parseAddress(record.recipient),
+          type: sudtScript,
+        },
+        data: utils.toBigUInt128LE(amount),
+      };
+      totalSudtOutputCapacity += sudtCapacity;
+      totalSudtOutputAmount += amount;
+      recipientCell.cell_output.capacity = `0x${sudtCapacity.toString(16)}`;
+      txSkeleton = txSkeleton.update('outputs', (outputs) => {
+        return outputs.push(recipientCell);
+      });
+    });
+
+    // collect sudt and add multisig cell inputs
+    const omniLockMultisigScript = parseAddress(getOmniLockMultisigAddress());
+    const searchKey = {
+      script: omniLockMultisigScript,
+      script_type: ScriptType.lock,
+      filter: {
+        script: sudtScript,
+      },
+    };
+
+    const searchedMultisigCells = await this.collector.collectSudtByAmountAndCapacity(
+      searchKey,
+      totalSudtOutputAmount,
+      totalSudtOutputCapacity + sudtCapacity,
+    );
+    const searchedMultisigCellsAmount = searchedMultisigCells
+      .map((cell) => utils.readBigUInt128LE(cell.data))
+      .reduce((a, b) => a + b, 0n);
+    const searchedMultisigCellsCapacity = searchedMultisigCells
+      .map((cell) => BigInt(cell.cell_output.capacity))
+      .reduce((a, b) => a + b, 0n);
+
+    // add multisigChangeCell if searchedMultisigCellsAmount === totalSudtOutputAmount then use ckb cell for change
+    let omniLockMultisigChangeCell: Cell;
+    if (searchedMultisigCellsAmount > totalSudtOutputAmount) {
+      omniLockMultisigChangeCell = {
+        cell_output: {
+          capacity: `0x0`,
+          lock: omniLockMultisigScript,
+          type: sudtScript,
+        },
+        data: utils.toBigUInt128LE(searchedMultisigCellsAmount - totalSudtOutputAmount),
+      };
+    } else if (searchedMultisigCellsAmount === totalSudtOutputAmount) {
+      omniLockMultisigChangeCell = {
+        cell_output: {
+          capacity: `0x0`,
+          lock: omniLockMultisigScript,
+        },
+        data: '0x',
+      };
+    } else {
+      throw new Error(
+        `multisig cell sudt amount not enough, need ${totalSudtOutputAmount.toString()}, found ${searchedMultisigCellsAmount.toString()}`,
+      );
+    }
+    omniLockMultisigChangeCell.cell_output.capacity = `0x${(
+      searchedMultisigCellsCapacity - totalSudtOutputCapacity
+    ).toString(16)}`;
+    txSkeleton = txSkeleton.update('inputs', (inputs) => {
+      return inputs.push(...searchedMultisigCells);
+    });
+    txSkeleton = txSkeleton.update('outputs', (outputs) => {
+      return outputs.push(omniLockMultisigChangeCell);
+    });
+
+    // add collector cell to pay tx fee
```

Comment:
use collector account to pay tx fee ***and supply capacity for recipient_sudt_cells***, that is to say, do not supply capacity for recipient_sudt_cells from committeeMultisigCells. ref: design_doc/example_for_unlockSudt

Reply:
unlock sudt capacity change and amount change calculation logic refactored according to design_doc/example_for_unlockSudt